### DISCUSSION TITLE
Custom AutoCompleteBox

### DISCUSSION THREAD

**Discussion by pyking-wang** [AvaloniaUI/Avalonia#11889] [Q&A] [answered]
I hope the AutoCompleteBox control can have a drop-down button to display all binding options, and each item in the drop-down list has a delete button. How can I get it? Thank you all.
![image](https://github.com/AvaloniaUI/Avalonia/assets/76192669/7cba6b29-b795-4ac2-9aed-5e2c0fa6a357)

**Comment by Flithor** [comment]
Try write a new template.
Refer to existing template(Find files like `AutoCompleteBox.xaml`)

**Comment by sccombp** [comment, marked as answer]
The first part of your question has a possible solution in the following discussion [here](https://github.com/AvaloniaUI/Avalonia/discussions/11533).

You can use the AutoCompleteBoxs ItemTemplate to customise the dropdown list to add the delete button.

```XML
    <AutoCompleteBox Grid.Column="1" Margin="10" Name="ACBBehaviour"
                     IsTextCompletionEnabled="True"
                     FilterMode="Contains"
                     Watermark="Attached Behaviour (down arrow or F4 opens dropdown when empty string ... also has clickable button)">
      <i:Interaction.Behaviors>
        <bh:AutoCompleteZeroMinimumPrefixLengthDropdownBehaviour/>
      </i:Interaction.Behaviors>
      <AutoCompleteBox.ItemTemplate>
        <DataTemplate>
          <DockPanel LastChildFill="True" Margin="2" >
            <Button Content="X" DockPanel.Dock="Right"/>
            <TextBlock Text="{Binding}" DockPanel.Dock="Left"/>
          </DockPanel>
        </DataTemplate>
      </AutoCompleteBox.ItemTemplate>
    </AutoCompleteBox>
```

![image](https://github.com/AvaloniaUI/Avalonia/assets/124018297/1076f767-022a-4be8-b67c-a8cc87f37de1)

**Reply by pyking-wang** [reply]
<i:Interaction.Behaviors>
        <bh:AutoCompleteZeroMinimumPrefixLengthDropdownBehaviour/>
Thank you for your reply. I am a novice in Avalonia. May I ask what ‘i’ and ‘bh’ refer to? thanks.

**Reply by sccombp** [reply]
Those are aliases.
In the linked example solution you can see them at the top of the MainWindow.axml file.

in this case "i" is an alias for "Interactivity" and "bh" is an alias for "Behaviours"

![image](https://github.com/AvaloniaUI/Avalonia/assets/124018297/0ca167b1-4fc5-40fc-b7a2-8b15264f04b6)

**Reply by pyking-wang** [reply]
I have encountered some problems, what is the possible reason，Perhaps StyleInclude is missing from App.axaml？
![image](https://github.com/AvaloniaUI/Avalonia/assets/76192669/33361dda-d35f-4ea8-9565-7ddb5fcb6ed8)

**Reply by timunie** [reply]
make sure the Avalonia version you use matches the one that AvaloniaBehaviors use.

**Reply by pyking-wang** [reply]
I think I am missing Avalonia.Xaml.Behaviors and Avalonia.Xaml.Interactivity, both NuGet. Thank you.

**Comment by buzzware** [comment]
I created this issue https://github.com/AvaloniaUI/Avalonia/issues/15416 - you might like to vote for it @pyking-wang  @sccombp
